typescript/infra/src/govern/HyperlaneAppGovernor.ts

```diff
@@ -164,19 +240,18 @@ export abstract class HyperlaneAppGovernor<
       return false;
     };
 
-    if (await transactionSucceedsFromSender(signerAddress)) {
+    if (await transactionSucceedsFromSender(chain, signerAddress)) {
       return SubmissionType.SIGNER;
     }
 
     // 2. Check if the call will succeed via Gnosis Safe.
     const safeAddress = this.checker.configMap[chain].owner;
-    if (!safeAddress) throw new Error(`Owner address not found for ${chain}`);
-    // 2a. Confirm that the signer is a Safe owner or delegate.
-    // This should implicitly check whether or not the owner is a gnosis
-    // safe.
 
-    // TODO: support for ICA governance coming soon
     if (typeof safeAddress === 'string') {
+      if (!safeAddress) throw new Error(`Owner address not found for ${chain}`);
```

Comment:
I dont think the type would ever be string and this throws

Reply:
fixed